newreply.php

```diff
@@ -630,13 +630,13 @@
 
 				// Lets see if this post is on the same page as the one we're viewing or not
 				// if it isn't, redirect us
-				if($perpage > 0 && (($postcounter) % $perpage) == 0)
+				if($mybb->settings['postsperpage'] > 0)
 				{
-					$post_page = ($postcounter) / $mybb->settings['postsperpage'];
+					$post_page = ceil(($postcounter + 1) / $mybb->settings['postsperpage']);
```

Comment:
Shouldn't this `$postcounter + 1` expression rather be simply `$postcounter`?

Reply:
Would the new post be accounted for? The `$postcounter = $thread['replies'] + 1` initialization would only add the thread's opening post.

Reply:
Ah, yep, I missed that.